File: realtime-gateway/src/foam.ts
import * as foam from '@foam-ai/node-opentelemetry';
import { ExpressInstrumentation } from '@opentelemetry/instrumentation-express';
import type { InstrumentationBase } from '@opentelemetry/instrumentation';

const isProduction = process.env.NODE_ENV === 'production';
const apiKey = `Bearer ${process.env.FOAM_API_KEY}`;

foam.init({
  serviceName: 'realtime-gateway',
  isProduction,
  apiKey,
  additionalInstrumentations: [new ExpressInstrumentation({}) as unknown as InstrumentationBase],
});

File: helply-api/src/foam.rb
# frozen_string_literal: true

require 'foam/opentelemetry'

Foam.init(
  service_name: 'helply-api',
  is_production: Rails.env.production?,
  api_key: "Bearer #{ENV['FOAM_API_KEY']}",
  instrumentations: [
    OpenTelemetry::Instrumentation::Rails,
    OpenTelemetry::Instrumentation::ActiveRecord,
    OpenTelemetry::Instrumentation::Sidekiq
  ]
)